src/main/java/frc/robot/subsystems/Turret.java

```diff
@@ -31,9 +31,14 @@ public Turret(){
    * Moves the Turret
    * @param speed the speed at which to set the turret motor to
    */
-
   public void set(double speed){
-    m_turretMotor.set(ControlMode.PercentOutput, speed);
+    if(getAngle() < Constants.TURRET_ROTATION_MIN && speed < 0){
+      stop();
+    } else if (getAngle() > Constants.TURRET_ROTATION_MAX && speed > 0){
+      stop();
+    } else {
+      m_turretMotor.set(ControlMode.PercentOutput, speed);
+    }
```

Comment:
Not a big deal, but this can be simplified by combining your two `stop()` cases. Personally, I'm also generally a fan of checking for the "affirmative" or "usual" case, and leaving the unusual case to the `else`, but that's just personal preference.

```
if ((getAngle() < Constants.TURRET_ROTATION_MIN && speed > 0.0) ||
    (getAngle() > Constants.TURRET_ROTATION_MAX && speed < 0.0)) {
  m_turretMotor.set(ControlMode.PercentOutput, speed);
} else {
  stop();
}
```

Reply:
+1 For Combining the two `stop()` cases, other than that this looks great